tools/scripts/update_apple_frameworks.sh

```diff
@@ -0,0 +1,191 @@
+#! /bin/bash
+
+# This script scans Xcode directores for unknown frameworks
+# and tries to add them as Kotlin/Native def files.
+# Note that some frameworks are unsupported (e.g. swift-only) and marked as .disabled.
+# Others may be supported and manual adjustment required (.attention_required).
+#
+# jq is required to run the script. It can be installed via `brew install jq`.
+#
+# Args:
+# 1. Platform name: ios, tvos, osx or watchos
+# 2. Path to Kotlin/Native sources.
+
+NATIVE_SRCDIR=$2
+
+case $1 in
+tvos*)
+  DEV_SDK=$(xcrun --show-sdk-path --sdk appletvos)
+  SIM_SDK=$(xcrun --show-sdk-path --sdk appletvsimulator)
+  OS_NAME="tvOS"
+  ;;
+ios*)
+  DEV_SDK=$(xcrun --show-sdk-path --sdk iphoneos)
+  SIM_SDK=$(xcrun --show-sdk-path --sdk iphonesimulator)
+  OS_NAME="iOS"
+  ;;
+watchos*)
+  DEV_SDK=$(xcrun --show-sdk-path --sdk watchos)
+  SIM_SDK=$(xcrun --show-sdk-path --sdk watchsimulator)
+  OS_NAME="watchOS"
+  ;;
+osx*)
+  DEV_SDK=$(xcrun --show-sdk-path)
+  SIM_SDK=$DEV_SDK
+  OS_NAME="macOS"
+  ;;
+*)
+  echo "Expected one of: osx ios watchos tvos. Got: $1"
+  exit 1
+esac
+
+DEFS=$NATIVE_SRCDIR/platformLibs/src/platform/$1
+FRAMEWORKS_DEV=$DEV_SDK/System/Library/Frameworks
+FRAMEWORKS_SIM=$SIM_SDK/System/Library/Frameworks
+
+DEFS_FILE=$(mktemp)
+FRAMEWORKS_DEV_FILE=$(mktemp)
+FRAMEWORKS_SIM_FILE=$(mktemp)
+FRAMEWORKS_COMMON_FILE=$(mktemp)
+ls $DEFS | grep .def | cut -d '.' -f 1 > $DEFS_FILE
+ls $FRAMEWORKS_DEV | grep .framework | cut -d '.' -f 1 > $FRAMEWORKS_DEV_FILE
+ls $FRAMEWORKS_SIM | grep .framework | cut -d '.' -f 1 > $FRAMEWORKS_SIM_FILE
+
+# Some frameworks might be available for device but not for framework, and vice versa.
+# Platform library should contain the common part.
+comm -12 $FRAMEWORKS_DEV_FILE $FRAMEWORKS_SIM_FILE > $FRAMEWORKS_COMMON_FILE
+ABSENT=$(comm -13 $DEFS_FILE $FRAMEWORKS_COMMON_FILE)
```

Comment:
Is this correct? If a framework is available in Swift on device, we should make it available in Kotlin on device too.